torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -471,7 +471,9 @@ TensorView* arange(Val* start, Val* end, Val* step, DataType dtype) {
     end = castOp(DataType::Double, end);
     step = castOp(DataType::Double, step);
   }
-  auto size = castOp(DataType::Int, ceilDiv(sub(end, start), step));
+  // Make sure no negative value is passed to ceilDiv as the device
+  // implementation of ceilDiv assumes positive inputs
+  auto size = castOp(DataType::Int, ceilDiv(abs(sub(end, start)), abs(step)));
```

Comment:
Actually, I'm not really sure what the right behavior should be when the signs of `end - start` and `step` are different, which I think is invalid. Do we need to do error checking? Doing so in device code could be very costly, so we probably would want to do on the host side before launching kernels. 

In any case, unless we want to check it on device, the above workaround should be fine.

Reply:
I think we do need to check it on the host side, but I am not sure how. Would it be a good idea to have both a `BinaryOpType::CeilDiv` and a `BinaryOpType::CeilDivMaybeNegative`, and we use `CeilDivMaybeNegative` for arange, and use `CeilDiv` for others?

Reply:
I don't know if we would need additional ceilDiv expr types. I think it should be possible to determine if it's guaranteed to be safe or not by looking at Expr nodes, and we could just do host-side checks of those ceilDiv exprs that are not determined to be safe.

That said, I don't think it's important at this point. At least for now, I think it'd be just reasonable if nvFuser works if a given program is correct but could result in undefined behavior if an unsound program is given.

Reply:
But, if it should be validated anyway, I think we would want to extend `KernelIrScanner` to mark potentially unsafe ceilDiv exprs. As I mentioned above, I think we should be able to just analyze the input vals of those exprs and see if they are composed of just non-negative values such as domain extents.

Reply:
A note is fine for now. We can revisit in the future if needed.